src/gainmap.c

```diff
@@ -660,6 +655,21 @@ avifResult avifRGBImageComputeGainMap(const avifRGBImage * baseRgbImage,
         }
     }
 
+    // Compute the headroom for the base and alterate images. The gainmap is the log-ratio of the
+    // higher-headroom (more HDR) image to the lower-headroom (less HDR) image. Reverse the gainmap
+    // if this is the case.
+    gainMapMetadata.baseHdrHeadroom = log2f(AVIF_MAX(baseMax, kEpsilon));
+    gainMapMetadata.alternateHdrHeadroom = log2f(AVIF_MAX(altMax, kEpsilon));
```

Comment:
Move lines 683-690 here to be with lines 658-662 as before. These lines "fill in the gain map's metadata,", so they should stay together.

Reply:
I disagree here. We populate `gainMapMetadata` all over the function. The offsets are computed up on line 603/605.

This is really the most natural place to populate `gainMapMetadata.alternateHdrHeadroom` and `gainMapMetadata.baseHdrHeadroom`. The reason for this is that
* Whether or not the gainmap is negated depends on whether or not depends on these values
* Whether or not the gainmap is negated affects how to compute `gainMapMinLog2` and `gainMapMaxLog2`.
* We can't populate `gainMapMetadata.gainMapMin` until we've computed `gainMapMinLog2`
So it would be fairly awkward to 

I've changed this function to no longer say "Fill in the gain map's metadata", but rather to say "Populate the gain map metdata's min and max fields", and I've separated it (just with a newline and comment) from the part that converts to fractions.

I also moved populating of `gainMapMetadata.useBaseColorSpace` to where `gainMapMetadata` is initialized, and removed the local `useBaseColorSpace` variable.

Let me know if this still feels unreasonable to you.

Reply:
Thank you for the explanation. Your rearrangement of the code and new comments make sense.